### PR TITLE
JAMES-1759 Testing a user existence

### PR DESCRIPTION
**Pull request by chibenwa**
Convenient when the input of user list is super large

### CODE REVIEW

File: server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/UserRoutes.java [hunk 1]
```diff
@@ -188,6 +209,17 @@ private String removeUser(Request request, Response response) {
         }
     }
 
+    private String userExist(Request request, Response response) throws UsersRepositoryException {
+        Username username = extractUsername(request);
+        if (userService.userExists(username)) {
+            response.status(HttpStatus.OK_200);
+            return Constants.EMPTY_BODY;
```
Comment (blackheaven) [inline server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/UserRoutes.java]: Same output?

File: server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/UserRoutes.java [hunk 2]
```diff
@@ -124,6 +128,23 @@ public void defineDeleteUser() {
...
     }
 
+    @HEAD
+    @Path("/{username}")
+    @ApiOperation(value = "Testing an user existence")
+    @ApiImplicitParams({
+        @ApiImplicitParam(required = true, dataType = "string", name = "username", paramType = "path")
+    })
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.OK_200, message = "OK. User is removed."),
```
Comment (trantienduchn) [inline server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/UserRoutes.java]: wrong message

File: server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/UserRoutes.java [hunk 3]
```diff
@@ -124,6 +128,23 @@ public void defineDeleteUser() {
...
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Invalid input user."),
+        @ApiResponse(code = HttpStatus.NOT_FOUND_404, message = "Invalid input user."),
```
Comment (trantienduchn) [inline server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/UserRoutes.java]: wrong message

File: server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/UserRoutes.java [hunk 4]
```diff
@@ -212,12 +212,11 @@ private String removeUser(Request request, Response response) {
     private String userExist(Request request, Response response) throws UsersRepositoryException {
         Username username = extractUsername(request);
         if (userService.userExists(username)) {
-            response.status(HttpStatus.OK_200);
             return Constants.EMPTY_BODY;
```
Comment (blackheaven) [inline server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/UserRoutes.java]: remaining?

File: server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/UserRoutesTest.java [hunk 5]
```diff
@@ -507,6 +515,26 @@ default void putShouldFailOnUnknownExceptionOnGetUpdateUser(UsersRepository user
...
+                .head(USERNAME_WITH_DOMAIN.asString())
+            .then()
+                .statusCode(HttpStatus.OK_200);
+        }
+
+        @Test
+        void headShouldReturnNotFoundWhenUserDoesNotExist() {
+            when()
+                .head(USERNAME_WITH_DOMAIN.asString())
+                .then()
```
Comment (remk) [inline server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/UserRoutesTest.java]: indent

File: src/site/markdown/server/manage-webadmin.md [hunk 6]
```diff
@@ -320,6 +321,21 @@ Response codes:
 
 Note: if the user exists already, its password will be updated.
 
+###Testing a user existence
+
+```
+curl -XHEAD http://ip:port/users/usernameToBeUsed
+```
+
+Resource name usernameToBeUsed representing valid users,
```
Comment (remk) [inline src/site/markdown/server/manage-webadmin.md]: Resource name "usernameToBeUsed" represents a valid user

**Comment by nvduc91** [comment]
Read it

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by Arsnael** [comment]
resolve deps issue...

test this please

**Comment by remk** [comment]
```
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR]   symbol:   class MimeMessage
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR]   location: class org.apache.james.util.MimeMessageUtil
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR] /james-project/server/container/util/src/main/java/org/apache/james/util/MimeMessageUtil.java:[37,34] cannot find symbol
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR]   symbol:   class MimeMessage
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR]   location: class org.apache.james.util.MimeMessageUtil
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR] /james-project/server/container/util/src/main/java/org/apache/james/util/MimeMessageUtil.java:[43,19] cannot find symbol
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR]   symbol:   class MimeMessage
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR]   location: class org.apache.james.util.MimeMessageUtil
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR] /james-project/server/container/util/src/main/java/org/apache/james/util/MimeMessageUtil.java:[47,19] cannot find symbol
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR]   symbol:   class MimeMessage
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR]   location: class org.apache.james.util.MimeMessageUtil
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR] /james-project/server/container/util/src/main/java/org/apache/james/util/MimeMessageUtil.java:[47,85] cannot find symbol
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR]   symbol:   class MessagingException
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR]   location: class org.apache.james.util.MimeMessageUtil
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR] /james-project/server/container/util/src/main/java/org/apache/james/util/MimeMessageUtil.java:[51,19] cannot find symbol
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR]   symbol:   class MimeMessage
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR]   location: class org.apache.james.util.MimeMessageUtil
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR] /james-project/server/container/util/src/main/java/org/apache/james/util/MimeMessageUtil.java:[51,73] cannot find symbol
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR]   symbol:   class MessagingException
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR]   location: class org.apache.james.util.MimeMessageUtil
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR] /james-project/server/container/util/src/main/java/org/apache/james/util/MimeMessageUtil.java:[55,19] cannot find symbol
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR]   symbol:   class MimeMessage
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR]   location: class org.apache.james.util.MimeMessageUtil
```

**Comment by Arsnael** [comment]
first error line is this:
```
[970152c02f4fa449f05f310e50c2cdb20f44bdbb] [ERROR] error reading /root/.m2/repository/org/checkerframework/checker-qual/2.0.0/checker-qual-2.0.0.jar; zip file is empty
```
not related to the PR

**Comment by Arsnael** [comment]
rebase on master

**Comment by chibenwa** [comment]
```
[781ba98386f7511134327ef7828c13481b9a8618] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project rabbitmq-jmap-draft-integration-testing: There was a timeout or other error in the fork -> [Help 1]
```

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**